### DISCUSSION TITLE
Add vpn to my current android project

### DISCUSSION THREAD

**Discussion by beko963** [schwabe/ics-openvpn#1575] [General]
Hello dears
Hope you are doing well

I am making a ndw application but I need to start it with built in VPN to make a successfull connection to firebase with some countries 

Is that possible ? if yes , please advice how can I do it ?

Thanks in advance

**Comment by schwabe** [comment]
If you base the application on my application your application must be open source (GPL). As alternative you could use the remoteExample that remotely uses my app from to connect/disconnect the VPN.

**Reply by beko963** [reply]
Thank you for your reply , so It is not possible to make my APK run VPN in background without being GPL ?

**Reply by schwabe** [reply]
You can write your app from scratch without using my source code or contact me privately to buy a license but that is not cheap since I rather want people to use open source than using a closed source app.

**Reply by beko963** [reply]
Thank you for your prompt reply
If i make my app GPL , how can I use your app to proxy the connection to firebase ?

**Reply by schwabe** [reply]
I have no experience with Firebase to be honest. I would suspect like with every other Android app.

**Reply by beko963** [reply]
Thank you sir , is there any documentation about how to link this project ot my own project ?
